File: segundo_semestre/Python/leccion_10/persona.py
class Persona:
    contador_personas = 0  # variable de clase

    @classmethod
    def generar_siguiente_valor(cls):
        cls.contador_personas += 1
        return cls.contador_personas

    def __init__(self, nombre, edad):
        self.id_persona = Persona.generar_siguiente_valor()
        self.nombre = nombre
        self.edad = edad

    def __str__(self):
        return f"Persona[id_persona={self.id_persona}, nombre={self.nombre}, edad={self.edad}]"


# Ejemplo de uso
if __name__ == "__main__":
    persona1 = Persona("Juan", 30)
    print(persona1)
    persona2 = Persona("Ana", 25)
    print(persona2)
    persona3 = Persona("Luis", 40)
    print(persona3)
    print(f"Cantidad de personas: {Persona.contador_personas}")

File: segundo_semestre/JavaScript/leccion_7/persona.js
/*
 clase 9
 9.2 Creación de la Clase Persona

*/
class persona {
  static contadorPersonas = 0;

  constructor(nombre, apellido, edad) {
    this._idPersona = ++persona.contadorPersonas;
    this._nombre = nombre;
    this._apellido = apellido;
    this._edad = edad;
  }

  get idPersona() {
    return this._idPersona;
  }
  get nombre() {
    return this._nombre;
  }
  set nombre(nombre) {
    this._nombre = nombre;
  }
  get apellido() {
    return this._apellido;
  }
  set apellido(apellido) {
    this._apellido = apellido;
  }
  get edad() {
    return this._edad;
  }
  set edad(edad) {
    this._edad = edad;
  }

  toString() {
    return `idPersona: ${this._idPersona}, nombre: ${this._nombre}, apellido: ${this._apellido}, edad: ${this._edad}`;
  }
}
